### PR TITLE
Add docs for transformations

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: site/content/en/docs/user-manual/command-reference/transform.md [hunk 1]
```diff
@@ -83,103 +83,164 @@ datum transform -t rename source-1 -- -e '|frame_(\d+)|\\1|'
...
+Replace 'pattern' with 'replacement':
+```bash
+datum transform -t rename -- -e '|pattern|replacement|'
+```
 
-datum transform -t split -- \
-  -t detection --subset train:.5 --subset val:.2 --subset test:.3
+Remove the `frame_` prefix from item ids:
+```bash
+datum transform -t rename -- -e '|frame_(\d+)|\1|'
```
Comment (IRDonch) [inline site/content/en/docs/user-manual/command-reference/transform.md:152]: If the goal is to remove a prefix, wouldn't it be simpler to do `|^frame_||`?

File: site/content/en/docs/user-manual/command-reference/transform.md [hunk 2]
```diff
@@ -189,36 +250,404 @@ There are five methods of sampling (the `-m/--method` option):
...
+- `-h`, `--help` (flag) - Show this help message and exit
+- `-s`, `--subset` (str; repeatable) - Subset mapping of the form: `src:dst`
+
+##### `remap_labels` <a id="remap_labels-transform"></a>
+
+Changes labels in the dataset.
+
+A label can be:
+- renamed (and joined with existing) -
+  when specified `--label <old_name>:<new_name>`
```
Comment (IRDonch) [inline site/content/en/docs/user-manual/command-reference/transform.md:468]:
```suggestion
  when `--label <old_name>:<new_name>` is specified
```

File: site/content/en/docs/user-manual/command-reference/transform.md [hunk 3]
```diff
@@ -189,36 +250,404 @@ There are five methods of sampling (the `-m/--method` option):
...
+- `--default` (one of: `keep`, `delete`) - Action for unspecified labels
+  (default: `keep`)
+
+Examples:
+Remove the `person` label (and corresponding annotations):
+```bash
+datum transform -t remap_labels -- -l person: --default keep
+```
+
+Rename `person` to `pedestrian` and `human` to `pedestrian`, join:
```
Comment (IRDonch) [inline site/content/en/docs/user-manual/command-reference/transform.md:493]: Join what?

File: site/content/en/docs/user-manual/command-reference/transform.md [hunk 4]
```diff
@@ -189,36 +250,404 @@ There are five methods of sampling (the `-m/--method` option):
...
+Align the source dataset labels to \[`person`, `cat`, `dog`\]:
+```
+datum transform -t project_labels -- -l person -l cat -l dog
+```
+
+Set dataset labels to {`person`, `cat`, `dog`}, remove others, add missing.
+Original labels (can be any): `cat`, `dog`, `elephant`, `human`
+New labels: `person` (added), `cat` (kept), `dog` (kept)
+``` bash
+datum transform -t project_labels -- -l person -l cat -l dog
```
Comment (IRDonch) [inline site/content/en/docs/user-manual/command-reference/transform.md:537]: Isn't this the same example twice?

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 4 above)
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md:584]: Right, removed

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 3 above)
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Updated

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 1 above)
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Updated

File: site/content/en/docs/user-manual/command-reference/transform.md [hunk 5]
```diff
@@ -83,103 +83,164 @@ datum transform -t rename source-1 -- -e '|frame_(\d+)|\\1|'
...
+  occluded segments of covered masks
+- [`bbox_value_decrement`](#bbox_value_decrement-transform) - Subtracts
+  1 from bbox coordinates
+
+##### `rename` <a id="rename-transform"></a>
+
+Renames items in the dataset. Supports regular expressions.
+The first character in the expression is a delimiter for
+the pattern and replacement parts. Replacement part can also
+contain `string.format` tokens with `item` object available.
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Added

File: datumaro/plugins/transforms.py [hunk 6]
```diff
@@ -267,6 +278,10 @@ def convert_shape(shape):
             id=shape.id, attributes=shape.attributes, group=shape.group)
 
 class Reindex(Transform, CliPlugin):
+    """
+    Assigns sequential indices to dataset items.
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/transforms.py]: Updated

File: datumaro/plugins/transforms.py [hunk 7]
```diff
@@ -32,6 +32,14 @@
 
 
 class CropCoveredSegments(ItemTransform, CliPlugin):
+    """
+    Sorts polygons and masks ("segments") according to `z_order`,
+    crops covered areas of underlying segments. If a segment is split
+    into several independent parts, produces the corresponding number of
+    separate annotations joined into a group. Produces polygons and masks
+    as they were originally.
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/transforms.py]: Updated

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 1 above)
Comment (IRDonch) [reply site/content/en/docs/user-manual/command-reference/transform.md]: The replacement part is now `\\1` - that seems wrong.

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 2 above)
Comment (IRDonch) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Please update the other list items correspondingly.

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (IRDonch) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Please also change "tokens" to "replacement fields". That's the official term used in the Python documentation.

File: datumaro/plugins/ndr.py [hunk 8]
```diff
@@ -28,10 +28,18 @@ class UnderSamplingMethod(Enum):
...
     """
-    Near-duplicated image removal.|n
-    |n
     Removes near-duplicated images in subset|n
     |n
+    Remove duplicated images from a dataset. Keep at most N resulting images.|n
+    |n
+    Available sampling options (the `-e` parameter):|n
+    - `random` - sample from removed data randomly|n
+    - `similarity` - sample from removed data with ascending|N
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/ndr.py:37]: Missing "similarity", added

File: datumaro/plugins/ndr.py [hunk 9]
```diff
@@ -28,10 +28,18 @@ class UnderSamplingMethod(Enum):
 
 class NDR(Transform, CliPlugin):
     """
-    Near-duplicated image removal.|n
-    |n
     Removes near-duplicated images in subset|n
     |n
+    Remove duplicated images from a dataset. Keep at most N resulting images.|n
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/ndr.py:33]: Fixed

File: datumaro/plugins/ndr.py [hunk 10]
```diff
@@ -28,10 +28,18 @@ class UnderSamplingMethod(Enum):
...
+    |n
+    Available sampling options (the `-e` parameter):|n
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/ndr.py:35]: Added

File: datumaro/plugins/sampler/relevancy_sampler.py [hunk 11]
```diff
@@ -19,6 +19,19 @@ class RelevancySampler(Transform, CliPlugin):
     Sampler that analyzes model inference results on the dataset |n
     and picks the best sample for training.|n
     |n
+    Creates a dataset from K the most hard items for a model. The whole dataset
+    or a single subset will be split into the `sampled` and `unsampled` subsets,
+    based on the model confidence The dataset **must** contain model confidence
+    values in the `scores` attributes of annotations.|n
+    |n
+    There are five methods of sampling (the `-m/--method` option):|n
+    - `topk` - Return the k with high uncertainty data|n
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/sampler/relevancy_sampler.py]: Updated

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Changed + added class info

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 1 above)
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Fixed

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 2 above)
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Done

File: datumaro/plugins/sampler/relevancy_sampler.py [hunk 12]
```diff
@@ -19,6 +19,19 @@ class RelevancySampler(Transform, CliPlugin):
...
     |n
+    Creates a dataset from the `-k/--count` hardest items for a model.
+    The whole dataset or a single subset will be split into the `sampled`
+    and `unsampled` subsets based on the model confidence.
+    The dataset **must** contain model confidence
+    values in the `scores` attributes of annotations.|n
+    |n
+    There are five methods of sampling (the `-m/--method` option):|n
+    - `topk` - Return the k items with high uncertainty data|n
+    - `lowk` - Return the k items with low uncertainty data|n
```
Comment (IRDonch) [inline datumaro/plugins/sampler/relevancy_sampler.py:30]:
```suggestion
    - `topk` - Return the k items with the highest uncertainty data|n
    - `lowk` - Return the k items with the lowest uncertainty data|n
```

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (IRDonch) [reply site/content/en/docs/user-manual/command-reference/transform.md]: You haven't fixed everything consistently: in the docstring the string class name is wrong, while in the documentation page the replacement fields are not referred to correctly.

File: datumaro/plugins/sampler/relevancy_sampler.py (hunk 12 above)
Comment (zhiltsov-max) [reply datumaro/plugins/sampler/relevancy_sampler.py]: Done

File: site/content/en/docs/user-manual/command-reference/transform.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply site/content/en/docs/user-manual/command-reference/transform.md]: Done

**Review by IRDonch — APPROVED** [review]

**Pull request merged**